### DISCUSSION TITLE
Could the sidebar sites get custom icons, or get the favicon from html and cache after load, instead of an icon provider like google?

### DISCUSSION THREAD

**Discussion by Lovasz-Akos** [Floorp-Projects/Floorp#821] [Feature Request]
![image](https://github.com/Floorp-Projects/Floorp/assets/55391231/f66e4026-206a-4125-854a-34375a8f8208)
I have my homeassistant pinned on my sidebar but my dns is not registered at google so the icon will never show up. I could register and fix this specific issue but it would be nice in the future to be able to atleast set icons to sidebar sites, maybe from the workspace icon set even

**Comment by Telmesomething** [comment]
I never login to Google unless I have to change something on my account over there, but it's primarily a cache problem. Clearing that and reloading the page usually fixes the issue. I have my settings setup the way you see them in the following screenshot. 
![settings for clearing cache and cookies](https://github.com/Floorp-Projects/Floorp/assets/33827636/e2f2c912-b564-474c-83a5-53e609c79130)

The menu to do this is located in Settings --> Privacy & Security and scroll down to "History". Enable "Customs settings for history" in the dropdown menu and click "Settings" for "Clear history when Floorp closes" and then checkmark the appropriate boxes and click OK.

**Reply by Lovasz-Akos** [reply]
this is not a caching issue, it's a my dns not being registered on google console so it doesn't download my favicon issue, thus I wanna sidestep the process and requested the feature for cusom icons

**Reply by Lovasz-Akos** [reply]
but same could be said for DDG since that's also an icon provider in floorp

**Reply by Telmesomething** [reply]
My custom DNS is not registered to Google either, but I don't have any problem with bookmarks icons. Maybe you can try using the free [Mullvad VPN DNS resolvers](https://mullvad.net/en/help/dns-over-https-and-dns-over-tls) to see if it makes any difference. 

There's an article on the subject over at [Ghacks](https://www.ghacks.net/2023/11/11/mullvads-public-encrypted-dns-servers-run-in-ram-now/) if you need any help configuring it

![bookmarks with icons](https://github.com/Floorp-Projects/Floorp/assets/33827636/db684f39-44d5-4a9a-aa11-07b04272fabd)

**Reply by Lovasz-Akos** [reply]
still not talking about bookmarks but sidebar sites....

**Comment by regularentropy** [comment]
@Lovasz-Akos https://github.com/Floorp-Projects/Floorp-core/pull/41
